Dockerfile

```diff
@@ -0,0 +1,25 @@
+# Usage:
```

Comment:
What is this doing here?

Reply:
I am stuck in Windows currently so I created the Dockerfile so that I could get all the dependencies for running `make` to install properly.  I have more updates that make it more useful that I will be pushing soon that utilizes the build caching and I updated the comments to include how to utilize volumes and bind mounts with run to keep preserve all the downloaded packages between runs and to allow the linter to update the local code that is outside the docker container.

However if you do not want me to add it I can just remove it.

What is nice about using docker for running the make actions is that if it is used it gives you an isolated container from the rest of your machine so that you can get consistent results.  Also using docker makes it easier for others to work with the Makefile as all they need to do is install Docker and the rest will just work.

Reply:
You cna remove it, this could be added to the docs though. Or one of the other fiber repos

Reply:
Also if you choose to keep it I can create helper scripts for running docker build and docker run to make it easier to use .